virttest/utils_misc.py

```diff
@@ -2818,199 +2818,314 @@ def get_uptime(session=None):
+    part_dict = {}
+    for dev in devs:
+        dev = dev.split()
+        if "disk" in dev:
+            disks_dict[dev[0]] = dev
+        if "part" in dev:
+            part_dict[dev[0]] = dev
     if not partition:
-        cmd = "%s | grep -v [0-9]$" % cmd
-    status, output = session.cmd_status_output(cmd)
-    if status != 0:
-        raise exceptions.TestFail("Get disks failed with output %s" % output)
-    return set(output.split())
+        for disk in disks_dict.keys():
+            if disk in ",".join(part_dict.keys()):
+                disks_dict.pop(disk)
+    else:
+        for part in part_dict.keys():
+            disks_dict[part] = part_dict[part]
+    get_info_cmd = "udevadm info -q property -p /sys/block/%s"
+    for kname in disks_dict.keys():
+        output = session.cmd_output_safe(get_info_cmd % kname)
+        serial = re.findall(r"(?<=ID_SERIAL=)(.*)", output, re.M)
+        if serial:
+            disks_dict.get(kname).extend(serial)
+        wwn = re.findall(r"(?<=ID_WWN=)(.*)", output, re.M)
+        if wwn:
+            disks_dict.get(kname).extend(wwn)
+    return disks_dict
 
 
-def get_all_disks_did(session, partition=False):
+def get_windows_disks_index(session, image_size):
     """
-    Get all disks did lists in a linux guest, each disk list
-    include disk kname, serial and wwn.
+    Get all disks index which show in 'diskpart list disk'.
+    except for system disk.
 
     :param session: session object to guest.
-    :param partition: if true, get all disks did lists; otherwise,
-                      get the ones with no partition.
-    :return: a dict with all disks did lists each include disk
-             kname, serial and wwn.
-    """
-    disks = list_linux_guest_disks(session, partition)
-    logging.debug("Disks detail: %s" % disks)
-    all_disks_did = {}
-    for line in disks:
-        kname = line.split('/')[2]
-        get_disk_info_cmd = "udevadm info -q property -p /sys/block/%s" % kname
-        output = session.cmd_output_safe(get_disk_info_cmd)
-        re_str = r"(?<=DEVNAME=/dev/)(.*)|(?<=ID_SERIAL=)(.*)|"
-        re_str += "(?<=ID_SERIAL_SHORT=)(.*)|(?<=ID_WWN=)(.*)"
-        did_list_group = re.finditer(re_str, output, re.M)
-        did_list = [match.group() for match in did_list_group if match]
-        all_disks_did[kname] = did_list
-
-    return all_disks_did
-
-
-def format_windows_disk(session, did, mountpoint=None, size=None,
+    :param image_size: image size. e.g. 40M
+    :return: a list with all disks index except for system disk.
+    """
+    disk_indexs = []
+    list_disk_cmd = "echo list disk > disk && "
+    list_disk_cmd += "echo exit >> disk && diskpart /s disk && del /f disk"
+    disks = session.cmd_output(list_disk_cmd)
+    size_type = image_size[-1] + "B"
+    if size_type == "MB":
+        disk_size = image_size[:-1] + " MB"
+    elif size_type == "GB" and int(image_size[:-1]) < 8:
+        disk_size = str(int(image_size[:-1]) * 1024) + " MB"
```

Comment:
I think it is not necessary to add a `"B"` on line 2874, you can check for `"M"` and `"G"`.
Not sure why you convert it to MB when it is less than 8G

Reply:
(1) I think add \"B\" is more clearer, easy to read code. 
(2) Not sure why you convert it to MB when it is less than 8G.  
    Because this is default in tool \"diskpart\"

Reply:
please add a comment for tool "diskpart"

Reply:
in diskpart:
if disk size < 8GB: it displays as MB
else: it displays as GB